src/mcts/search.cc

```diff
@@ -1466,7 +1441,88 @@ void SearchWorker::DoBackupUpdateSingleNode(
   search_->total_playouts_ += node_to_process.multivisit;
   search_->cum_depth_ += node_to_process.depth * node_to_process.multivisit;
   search_->max_depth_ = std::max(search_->max_depth_, node_to_process.depth);
-}  // namespace lczero
+}
+
+bool SearchWorker::MaybeConvertParent(Node* p, float m) const {
+  auto found_tb = false;
+  auto losing_m = 0.0f;
+
+  // Determine the maximum (Lower, Upper) bounds across all children.
+  // (-1,-1) Loss (initial and lowest bounds)
+  // (-1, 0) Can't Win
+  // (-1, 1) Regular node
+  // ( 0, 0) Draw
+  // ( 0, 1) Can't Lose
+  // ( 1, 1) Win (highest bounds)
+  auto max_L = -1;
+  auto max_U = -1;
+  for (const auto& edge : p->Edges()) {
```

Comment:
I haven't rewritten this yet, but I would think with each node specifying its `canWin` `canDraw` and `canLose`, there shouldn't need to be a check for terminal/cantwin/etc., and there should be some simple accumulation. Although we probably still need some branching to support tb-terminal-ness.

Reply:
Rewrote with enum bounds saved in the node to then max directly.